x-pack/plugins/security_solution/server/utils/with_security_span.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SpanOptions, withSpan } from '@kbn/apm-utils';
+import agent from 'elastic-apm-node';
+
+type Span = Exclude<typeof agent.currentSpan, undefined | null>;
+
+export const withSecuritySpan = <T>(
```

Comment:
nit: Add JSDoc for when folks should use this function and any necessary pre-req's (does this need to happen within the scope of `agent.setTransactionName()`?)

Reply:
Added doc. As for prerequisites, there aren't any. We can use this method anywhere throughout our codebase. All main code paths are already wrapped in transactions on the framework level.